xcm/xcm-builder/src/nonfungibles_v2_adapter.rs

```diff
@@ -0,0 +1,438 @@
+		AccountId,
+		CheckAsset,
+		CheckingAccount,
+		ItemConfig,
+	>
+{
+	fn can_check_in(origin: &MultiLocation, what: &MultiAsset, context: &XcmContext) -> XcmResult {
+		NonFungiblesV2MutateAdapter::<
+			Assets,
+			Matcher,
+			AccountIdConverter,
+			AccountId,
+			CheckAsset,
+			CheckingAccount,
+			ItemConfig,
+		>::can_check_in(origin, what, context)
+	}
+
+	fn check_in(origin: &MultiLocation, what: &MultiAsset, context: &XcmContext) {
+		NonFungiblesV2MutateAdapter::<
+			Assets,
+			Matcher,
+			AccountIdConverter,
+			AccountId,
+			CheckAsset,
+			CheckingAccount,
+			ItemConfig,
+		>::check_in(origin, what, context)
+	}
+
+	fn can_check_out(dest: &MultiLocation, what: &MultiAsset, context: &XcmContext) -> XcmResult {
+		NonFungiblesV2MutateAdapter::<
+			Assets,
+			Matcher,
+			AccountIdConverter,
+			AccountId,
+			CheckAsset,
+			CheckingAccount,
+			ItemConfig,
+		>::can_check_out(dest, what, context)
+	}
+
+	fn check_out(dest: &MultiLocation, what: &MultiAsset, context: &XcmContext) {
+		NonFungiblesV2MutateAdapter::<
+			Assets,
+			Matcher,
+			AccountIdConverter,
+			AccountId,
+			CheckAsset,
+			CheckingAccount,
+			ItemConfig,
+		>::check_out(dest, what, context)
+	}
+
+	fn deposit_asset(what: &MultiAsset, who: &MultiLocation, context: &XcmContext) -> XcmResult {
+		NonFungiblesV2MutateAdapter::<
+			Assets,
+			Matcher,
+			AccountIdConverter,
+			AccountId,
+			CheckAsset,
+			CheckingAccount,
+			ItemConfig,
+		>::deposit_asset(what, who, context)
+	}
+
+	fn withdraw_asset(
+		what: &MultiAsset,
+		who: &MultiLocation,
+		maybe_context: Option<&XcmContext>,
+	) -> result::Result<xcm_executor::Assets, XcmError> {
+		NonFungiblesV2MutateAdapter::<
+			Assets,
+			Matcher,
+			AccountIdConverter,
+			AccountId,
+			CheckAsset,
+			CheckingAccount,
+			ItemConfig,
+		>::withdraw_asset(what, who, maybe_context)
+	}
+
+	fn transfer_asset(
+		what: &MultiAsset,
+		from: &MultiLocation,
+		to: &MultiLocation,
+		context: &XcmContext,
+	) -> result::Result<xcm_executor::Assets, XcmError> {
+		NonFungiblesV2TransferAdapter::<Assets, Matcher, AccountIdConverter, AccountId>::transfer_asset(
+			what, from, to, context,
+		)
+	}
+}
+
+#[derive(
+	Copy, Clone, Decode, Encode, Eq, PartialEq, Ord, PartialOrd, Debug, TypeInfo, MaxEncodedLen,
+)]
+pub struct MultiLocationCollectionId(MultiLocation);
+impl MultiLocationCollectionId {
+	/// Consume `self` and return the inner MultiLocation.
+	pub fn into_inner(self) -> MultiLocation {
+		self.0
+	}
+
+	/// Return a reference to the inner MultiLocation.
+	pub fn inner(&self) -> &MultiLocation {
+		&self.0
+	}
+}
+
+impl Incrementable for MultiLocationCollectionId {
+	fn increment(&self) -> Option<Self> {
+		None
+	}
+
+	fn initial_value() -> Option<Self> {
+		None
+	}
+}
+
+impl From<MultiLocation> for MultiLocationCollectionId {
+	fn from(value: MultiLocation) -> Self {
+		MultiLocationCollectionId(value)
+	}
+}
+
+impl Into<MultiLocation> for MultiLocationCollectionId {
```

Comment:
You should always implement `From` and not `Into`: https://www.bradcypert.com/rust-from-into/#two-traits-which-to-implement